### PR TITLE
(fix) : fetch claims from API

### PR DESCRIPTION
**Pull request by amosmachora**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
This PR now fetches live claim information from the API provided by @Injiri . 

## Screenshots
image

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by Injiri** [comment]
@amosmachora  we add **externalId** on the status for the requests posted.

File: packages/esm-billing-app/src/claims/claims-management/table/claims-list-table.component.tsx [hunk 1]
```diff
@@ -1,70 +1,115 @@
...
 import styles from './claims-list-table.scss';
-import { Claimrows, preAuthRows } from './table-mock-up-data';
-import useShaData from './hook/table.resource';
+import { useFacilityClaims } from './use-facility-claims';
 
 const ClaimsManagementTable: React.FC = () => {
+  const { claims, isLoading } = useFacilityClaims();
+  const [pageSize, setPageSize] = useState(5);
+  const { paginated, goTo, results, currentPage } = usePagination(claims, pageSize);
+  const { pageSizes } = usePaginationInfo(pageSize, claims.length, currentPage, results.length);
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/claims-management/table/claims-list-table.component.tsx:24]: We now have a hook from `@openmrs/esm-framework` that allows us to use backend pagination to try save and improve performance. We should have that as TODO as a follow up on this PR

File: packages/esm-billing-app/src/claims/claims-management/table/claims-list-table.component.tsx [hunk 2]
```diff
@@ -1,70 +1,115 @@
...
+  const responsiveSize = isDesktop(useLayoutType()) ? 'sm' : 'lg';
   const { t } = useTranslation();
-  const shaData = useShaData();
 
   const headers = [
-    { key: 'visitTime', header: t('visitTime', 'Visit Time') },
-    { key: 'patientName', header: t('patientName', 'Patient Name') },
-    { key: 'claimCode', header: t('claimCode', 'Claim Code') },
-    { key: 'status', header: t('status', 'Status') },
+    { key: 'claimCode', header: 'Claim Code' },
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/claims-management/table/claims-list-table.component.tsx:29]:
```suggestion
    { key: 'claimCode', header: t('claimCode', Claim Code') },
```

File: packages/esm-billing-app/src/claims/claims-management/table/use-facility-claims.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { FetchResponse, openmrsFetch, restBaseUrl } from '@openmrs/esm-framework';
+import useSWR from 'swr';
+
+export type FacilityClaim = {
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/claims-management/table/use-facility-claims.ts:4]: type definitions should be centralized in `types` file

File: packages/esm-billing-app/src/claims/claims-management/table/use-facility-claims.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          ...claim,
+          providerName: claim.provider?.display,
+          approvedTotal: claim.approvedTotal ?? 0,
+          status:
+            claim.status !== 'ENTERED' && claim.status !== 'ERRORED'
+              ? `${claim.status} - ${claim.externalId}`
+              : claim.status,
+          id: claim.uuid,
+        };
+      }) ?? [],
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/claims/claims-management/table/use-facility-claims.ts:38]:
```tsx
  const formatClaimStatus = (status: ClaimStatus, externalId: string | null): string => {
    const shouldAppendExternalId = status !== 'ENTERED' && status !== 'ERRORED';
    return shouldAppendExternalId && externalId ? `${status} - ${externalId}` : status;
  };

  const formatClaim = (claim: FacilityClaimResponse): FormattedFacilityClaim => ({
    ...claim,
    id: claim.uuid,
    providerName: claim.provider?.display,
    approvedTotal: claim.approvedTotal ?? 0,
    status: formatClaimStatus(claim.status, claim.externalId)
  });

  const formattedClaims = data?.data.results.map(formatClaim) ?? [];
  ```
  
  Then simply have a simple return from the hook
  ```tsx
   return {
    claims: formattedClaims,
    error: error,
    isLoading,
    isValidating,
    mutate,
  };
```

**Review by donaldkibet — APPROVED** [review]

File: packages/esm-billing-app/src/claims/claims-management/table/claims-list-table.component.tsx (hunk 1 above)
Comment (amosmachora) [reply packages/esm-billing-app/src/claims/claims-management/table/claims-list-table.component.tsx:24]: Great. Thanks for highlighting that. I didn`t know that.

Comment (amosmachora) [reply packages/esm-billing-app/src/claims/claims-management/table/claims-list-table.component.tsx:24]: Now that i think of it maybe there should be a project to migrate all `usePagination` hooks to use the new backend pagination hook. What do you think @donaldkibet .

**Comment by ojwanganto** [comment]
Will we have filters for dates i.e. claims for January, Feb, etc? Are there any guidance on the implementation approach?

**Comment by ojwanganto** [comment]
We'll find time to discuss the designs. No worries, and thanks for the good work.

**Pull request merged**